### PR TITLE
many: at seeding try to capture cloud information into core config under "cloud"

### PR DESCRIPTION
**Pull request by pedronis**
This is done from the running of the internal core configure code at that time.

### CODE REVIEW

File: overlord/configstate/configcore/corecfg.go [hunk 1]
```diff
@@ -57,6 +58,11 @@ func Run(tr Conf) error {
 		return err
 	}
 
+	// capture cloud information
+	if err := handleCloud(tr); err != nil {
```
Comment (mvo5) [inline overlord/configstate/configcore/corecfg.go]: (nitpick) I wonder if a more specific name is more appropriate here given that this function only acts when seeding. Maybe something like `setCloudConfWhenSeeding` or similar?

File: overlord/devicestate/firstboot_test.go [hunk 2]
```diff
@@ -169,18 +181,91 @@ func (s *FirstBootTestSuite) TestPopulateFromSeedOnClassicNoSeedYaml(c *C) {
...
+	checkTrivialSeeding(c, tsAll)
 
 	ds, err := auth.Device(st)
 	c.Assert(err, IsNil)
 	c.Check(ds.Brand, Equals, "my-brand")
 	c.Check(ds.Model, Equals, "my-model-classic")
 }
 
+func (s *FirstBootTestSuite) TestPopulateFromSeedOnClassicNoSeedYamlWithCloudInstanceData(c *C) {
+	release.OnClassic = true
```
Comment (mvo5) [inline overlord/devicestate/firstboot_test.go]: Should we use `restore := release.MockOnClassic(true); defer restore()` here?

File: dirs/dirs.go [hunk 3]
```diff
@@ -224,6 +225,7 @@ func SetRootDir(rootdir string) {
 	SystemApparmorCacheDir = filepath.Join(rootdir, "/etc/apparmor.d/cache")
 
 	CloudMetaDataFile = filepath.Join(rootdir, "/var/lib/cloud/seed/nocloud-net/meta-data")
```
Comment (chipaca) [inline dirs/dirs.go]: what uses this? (grep says nothing) is it related to this work?

File: overlord/devicestate/firstboot_test.go (hunk 2 above)
Comment (chipaca) [reply overlord/devicestate/firstboot_test.go]: the test suite is already doing the restore, although a comment would be nice

File: dirs/dirs.go (hunk 3 above)
Comment (pedronis) [reply dirs/dirs.go]: that's probably a question for @mvo5, I know image does some cloud-init related stuff, but I don't know if this one is 15.04 leftover

File: overlord/devicestate/firstboot_test.go (hunk 2 above)
Comment (pedronis) [reply overlord/devicestate/firstboot_test.go]: tbh, I just followed the style of some tests already present here

File: dirs/dirs.go (hunk 3 above)
Comment (chipaca) [reply dirs/dirs.go]: tbh that's probably fine then :-)

File: overlord/configstate/configcore/corecfg.go (hunk 1 above)
Comment (pedronis) [reply overlord/configstate/configcore/corecfg.go]: using setCloudInfoWhenSeeding

File: dirs/dirs.go (hunk 3 above)
Comment (mvo5) [reply dirs/dirs.go]: This looks very much like a leftover from the old days. Fwiw, `git blame` blames @chipaca for touching it last (8759e4074e92386d33851278aff3fd31cbab27f9 from 2015) - the commit message is a bit obscure though.

**Pull request merged**